controllers/cloud.redhat.com/clowdenvs.go

```diff
@@ -0,0 +1,111 @@
+package controllers
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	clowder "github.com/RedHatInsights/clowder/apis/cloud.redhat.com/v1alpha1"
+	"github.com/go-logr/logr"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func CreateClowdEnv(ctx context.Context, cl client.Client, spec clowder.ClowdEnvironmentSpec, nsName string) error {
+	env := clowder.ClowdEnvironment{
+		Spec: spec,
+	}
+	env.SetName(fmt.Sprintf("env-%s", nsName))
+	env.Spec.TargetNamespace = nsName
+
+	ns, err := GetNamespace(ctx, cl, nsName)
+	if err != nil {
+		return err
+	}
+
+	env.SetOwnerReferences([]metav1.OwnerReference{
+		{
+			APIVersion: ns.APIVersion,
+			Kind:       ns.Kind,
+			Name:       ns.Name,
+			UID:        ns.UID,
+		},
+	})
+
+	if err := cl.Create(ctx, &env); err != nil {
```

Comment:
UpdateOrErr (CreateOrErr?) from clowder

Reply:
Adding this as a ticket (potentially - this create is always called with a unique name so the check may not be necessary)